Sources/SupabaseStorage/StorageError.swift

```diff
@@ -1,17 +1,42 @@
 import Foundation
 
-public struct StorageError: Error {
+public struct StorageError: Error, Codable {
   public var statusCode: Int?
+
   public var message: String?
+  public var error: String?
 
-  public init(statusCode: Int? = nil, message: String? = nil) {
+  public init(
+    statusCode: Int? = nil,
+    message: String? = nil,
+    error: String? = nil
+  ) {
     self.statusCode = statusCode
     self.message = message
+    self.error = error
+  }
+
+  enum CodingKeys: CodingKey {
+    case statusCode
+    case message
+    case error
+  }
+
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    if let statusCodeInt = try? container.decodeIfPresent(Int.self, forKey: .statusCode) {
+      statusCode = statusCodeInt
+    } else if let statusCodeString = try container.decodeIfPresent(String.self, forKey: .statusCode) {
+      statusCode = Int(statusCodeString)
+    }
```

Comment:
Does this actually get returned as both `Int` and `String` depending on the endpoint?

Reply:
oops, nope. that's a leftover